docs/design-notes/qualifying-replicas-for-reads.md

```diff
@@ -0,0 +1,83 @@
+# Qualifying replicas for read operations
+
+## Intro
+
+When a read request arrives at the coordinator node, it is later
+forwarded to one or more replicas in order to read the actual data
+and return it back to the client. Each request has an associated
+timeout, which indicates how long the client is willing to wait
+for the query results.
+This timeout information can be leveraged to lift some pressure off
+the coordinator if some of the target replicas are in fact dead,
+overloaded to the point of unresponsiveness, having network problems, etc.
+
+## Gathering last seen information
+
+The coordinator should be able to predict whether a replica is likely
+to respond to a request in time or not. This prediction should be as
+precise as possible, but at the same time have minimal overhead, because
+performing read requests is on the hot path. A simple failure detection
+mechanism with high granularity is implemented as follows:
+ * each time a read request is sent to a replica, this attempt's timestamp
+   is recorded
+ * first timestamp of a streak of unresponded requests is recorded separately
```

Comment:
I'm not sure what this streak means, or how you can calculate it, maybe you can explain this better. Imagine in times 1,2,3...100 you send 100 requests, and until now requests 2 through 70 were responded. What is the "first timestamp of a streak of unresponded requests" - is it time 1? 71? or what?

Reply:
(I hope I still remember and understand my original idea...) 71 - what I meant to express is that if there's a series of requests that didn't get any response, I remember the timestamp of the first one. Here's an example for time points 1 to 8:
1. send A
2. send B
3. recv A
4. recv B
5. send C <---- this is the timestamp I'm interested in (from the point of view of time point 8. below); that's when we stopped getting any responses
6. send D
7. send E
8. \<now>